.upstream-tests/test/cuda/memory_resource/memory_resource.concepts/has_property.pass.cpp

```diff
@@ -0,0 +1,60 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2022 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+//===----------------------------------------------------------------------===//
+
+// UNSUPPORTED: c++03, c++11
+
+// cuda::mr::has_property, cuda::mr::has_property_with
+#include <cuda/memory_resource>
+
+struct prop_with_value {
+  using value_type = int;
+};
+struct prop {};
+
+static_assert(cuda::mr::property_with_value<prop_with_value>);
+static_assert(!cuda::mr::property_with_value<prop>);
```

Comment:
I just noticed that the property machinery is in the `cuda::mr::` namespace. @griwes correct me if I'm wrong, but I think we want this to just be in the `cuda::` namespace as we have plans to reuse properties for things outside of just memory resources.

Reply:
Moved to the cuda namespace